stdlib.sh

```diff
@@ -1068,6 +1068,71 @@ layout_pyenv() {
   [[ -n "$PYENV_VERSION" ]] && export PYENV_VERSION
 }
 
+# Usage: layout uv [<python_version>]
+#
+# Uses uv to create and a virutal environment, and then activates it.
+# See https://docs.astral.sh/uv/pip/environments/ for more details.
+# If no python version is specified, the default python version is used. uv will
+# also install the requested python version if it is not installed.
+layout_uv() {
+    # optional argument to specify python version
+    local python_version=$1
+    if [[ -d ".venv" ]]; then
+        VIRTUAL_ENV="$(pwd)/.venv"
+    fi
+
+    if [[ -z $VIRTUAL_ENV || ! -d $VIRTUAL_ENV ]]; then
+        log_status "No virtual environment exists. Executing \`uv venv\` to create one."
+        # If there is a python version specified, use it to create the environment
+        if [[ -n $python_version ]]; then
+            uv venv --python "$python_version"
+        else
+            uv venv
+        fi
```

Comment:
```suggestion
        uv venv "$@"
```
It might be simpler to forward all the arguments to `uv`. This would also allow removing the `layout_uvp` function.

Reply:
That would be simpler, but I like that this method lines up more closely with my previous favorite layout, `layout pyenv` which also takes a python version. In this case, `--python` is the main argument which is actually useful with `uv venv`.

Although I would love to get away from having `layout_uv` vs `layout_uvp`, I'm not sure how to avoid that practically, since the base command for `uv` is `uv venv`, and for `uvp` it's `uv init`. I suppose we could do `uv "$@"`, but I really like how `layout uv` and `layout uvp` just work without any extra args needed. Otherwise, you would always have to do something like `layout uv venv`.